### PR TITLE
Add an option to specify "Operating System Profile" when creating machine deployments

### LINKED ISSUE #4580: Add an option to specify "Operating System Profile" when creating machine deployments
### What would you like to be added? / User Story
As a user, I'd like an option to specify "Operating System Profile" when creating machine deployments.

### Requirements/Details

1. This option should not be included in the edit view for machine deployments. But viewing the selected OSP is required in the edit/details view.
2. Only display this option if OSM feature gate is enabled on the seed AND cluster level.
3. We don't need to display any default value for this. So a note along these lines would be nice: 
`Use this field to override the default operating system profile used for this machine deployment.`
4. From an implementation perspective, API changes should not be required IMO. `nodeDeployment` already includes annotations so we just have to add the key `k8c.io/operating-system-profile` with the value specified in the UI.
5. DON'T add this `nodeDeployment.annotations` in the payload if the value provided was null/empty.

### Screenshots for reference

Screenshot 2022-06-10 at 6 33 06 PM

Screenshot 2022-06-10 at 6 39 56 PM

### Acceptance criteria

Users can choose which OSP to use as default for a machine deployment.

### PR DESCRIPTION
**Pull request by ahmadhamzh**
### What this PR does / why we need it
adding new field to add operating system profile when creating new MD  
### Which issue(s) this PR fixes
"closes #4580"

```release-note
Allow the user to specify the operating system profile when creating machine deployment.
```

### CODE REVIEW

File: src/app/core/services/node.ts
Comment (KhizerRehan) [inline src/app/core/services/node.ts:100]: can we define constant for `k8c.io/operating-system-profile`? in common  place so that can be used in both file.

File: src/app/node-data/component.ts
Comment (KhizerRehan) [inline src/app/node-data/component.ts:53]: Avoid spacing in control name see above controls. Use `camelCasing` syntax
e.g`RhelSubscriptionManagerPassword,OperatingSystem`

**Review by KhizerRehan — CHANGES_REQUESTED** [review]
kindly, see comments

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]
Please display the select operating system profile, if overridden, in the cluster summary page as well.

File: src/app/node-data/component.ts
Comment (KhizerRehan) [inline src/app/node-data/component.ts:108]: use single `_` underscore

File: src/app/node-data/template.html
Comment (ahmedwaleedmalik) [reply src/app/node-data/template.html]: @ahmadhamzh this sentence should end with a full stop "."

File: src/app/node-data/template.html
Comment (ahmedwaleedmalik) [inline src/app/node-data/template.html:178]:
```suggestion
      <mat-hint *ngIf="!dialogEditMode">Leave this field blank to use default operating system profile.</mat-hint>
```

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

**Comment by ahmedwaleedmalik** [comment]
LGTM although you'll need approval from `sig-ui` members to move forward with this PR. Also, would be super awesome if you can rebase your branch and probably just have one meaningful commit ;) But I'll leave that up to you.

**Comment by Waseem826** [comment]
One issue I noticed:
- Select (check) Operating System Manager on cluster step
- Go to Initial Nodes step
- Enter some value in Operating System Profile
- Go back to cluster step
- Uncheck Operating System Manager
- Operating System Profile value is not cleared and shown on summary step
- Annotation is also sent in API payload

screenshot-localhost_8000-2022 06 21-13_11_20

**Comment by ahmadhamzh** [comment]
i just fix the problem but according to where it should be display i think inside the initial node summary and it call operating system profile.
so the same place but change operating system manager to operating system profile 
right ?
@ahmedwaleedmalik

File: src/app/shared/components/cluster-summary/template.html
Comment (Waseem826) [inline src/app/shared/components/cluster-summary/template.html]:
```suggestion
            <div key>Operating System Profile</div>
```

File: src/app/wizard/component.ts
Comment (Waseem826) [inline src/app/wizard/component.ts:45]:
```suggestion
  operatingSystemProfileAnnotation = OPERATING_SYSTEM_PROFILE_ANNOTATION;
```

File: src/app/wizard/step/cluster/component.ts
Comment (Waseem826) [inline src/app/wizard/step/cluster/component.ts:105]:
```suggestion
  operatingSystemManagerFeatureEnabled: boolean;
```

**Review by Waseem826 — CHANGES_REQUESTED** [review]

File: src/app/shared/components/cluster-summary/template.html
Comment (ahmadhamzh) [reply src/app/shared/components/cluster-summary/template.html]: changed in the last commit
https://github.com/kubermatic/dashboard/pull/4602/commits/0fece656f1bf38e9726c31e5e484348540fa9884

File: src/app/core/services/node.ts
Comment (Waseem826) [inline src/app/core/services/node.ts:107]: I think we should add a null check here on `md.annotations` 
```suggestion
          operatingSystemProfile: md.annotations?.[OPERATING_SYSTEM_PROFILE_ANNOTATION],
```

**Review by Waseem826 — CHANGES_REQUESTED** [review]

**Comment by Waseem826** [comment]
@ahmadhamzh I think we should add a release note since this is something new being added.

**Review by Waseem826 — APPROVED** [review]

**Pull request merged**
